src/components/organism/ToDoList/ToDoList.tsx

```diff
@@ -1,79 +1,152 @@
-            const todoitemOnDelete = () =>{ setTargetToDo(todo);setDeletedrawer(false)}
-            const todoitemOnUpdate = () =>{ Toast({message:"To Do Updated."}); UpdateCallback()}
+        }
+        else {
+            //sorts between complete and uncomplete >
+            const uncompletedtodos = reduxToDos.filter(currentTodo =>{return !currentTodo.complete})
+            const completedtodos = reduxToDos.filter(currentTodo =>{return currentTodo.complete})
+            sortedToDos = [...uncompletedtodos,...completedtodos]
             //<
-            return <ToDoItem 
-            onSelect={()=>todoitemOnSelect()} 
-            selected={selectedToDos.includes(todo)} 
-            selectionMode={selectionMode} 
-            key={todo.id} 
-            value={todo.todotext} 
-            completed={todo.complete}
-            onTextClick={()=>{todoitemOnTextClick()}}
-            onDelete={()=>{todoitemOnDelete()}}
-            onUpdate={()=>{todoitemOnUpdate()}}/>
         }
-
+    //<
+    //>
+    const isFilled = sortedToDos.length;
+    const selectedAll = sortedToDos === selectedToDos;
+    const noReduxToDos = !(reduxToDos.length > 0);
+    //<
+    //Additional helper func>
+    useEffect(()=>{setSelectedToDos([])},[selectionMode])
+    const onYes = ()=> {
+        Toast({message:"To Do Deleted."}); 
+        dispatch(deletetodo(targetToDo)); 
+    }
+    const onDeleteAllSelected = () =>{
+        Toast({message:"To Dos Deleted."}); 
+        dispatch(deletetodos([...selectedToDos])); 
+    }
+    const onCompleteAllSelected = () =>{
+        Toast({message:"To Dos Completed."});
+        dispatch(completetodos([...selectedToDos])); 
+    }
+    const onSelectAll = () =>{
+        if(selectedAll){
+            setSelectedToDos([])
+        } else {
+            setSelectedToDos([...sortedToDos])
+        }
+    }
+    //<       
   return (
-    <ToDoListContainer>
-        <ToDoListList>
-            {
-                uncompletedtodos.map((todo: todoprops) =>{ return todoitem(todo)})
-            }
-            { selectionMode? "":
-                completedtodos.map((todo: todoprops) =>{ return todoitem(todo)}) 
-            }
-        </ToDoListList>
-            <DeleteSelectedToDoDrawer 
-            onDeleteSelected={()=>{SDrawerOnDeleteSelected()}} 
-            onCompleteSelected={()=>{SDrawerOnCompleteSelected()}} 
-            onSelectAll={()=>{setSelectedToDos([...todos])}} 
-            hide={()=>setSelectedToDos([])} 
-            hidden={selectedToDos.length===0}/>
-            <DeleteToDoDrawer 
-            onYes={()=> {DrawerOnYes()}} 
-            hide={()=>{setDeletedrawer(true)}} 
-            hidden={deletedrawer}/>
-    </ToDoListContainer>
+    <Container>
+    {isFilled?
+    <>
+        <ToDoListContainer className={isFilled? "": "NoToDo"}>
+            <ToDoListList>
+                {sortedToDos.map((todo: todoprops) =>
+                { 
+                return PreparedTodoitem(todo,
```

Comment:
Sorry. I think there was a misunderstanding. Can you revert back your code here? You only need to move the functions inside `todoitem = (todo: todoprops)` outside from there. The final output should look something like this:
```
const ToDoList = ({selectionMode,UpdateCallback,searchedTodos,addToDo,search}: ToDoListProps) => {
        ...
        const todoitemOnSelect = (todo: todoprops) => {...}
        const todoitemOnTextClick = (todo: todoprops) => {...}
        const todoitemOnDelete = (todo: todoprops) =>{...}
        const todoitemOnUpdate = (todo: todoprops) =>{...}

        const todoitem = (todo: todoprops) => {
            return <ToDoItem 
            onSelect={()=>todoitemOnSelect(todo)} 
            selected={selectedToDos.includes(todo)} 
            selectionMode={selectionMode} 
            key={todo.id} 
            value={todo.todotext} 
            completed={todo.complete}
            onTextClick={()=>{todoitemOnTextClick(todo)}}
            onDelete={()=>{todoitemOnDelete(todo)}}
            onUpdate={()=>{todoitemOnUpdate(todo)}}/>
        }
        
        return (<Container>
            ...
            <ToDoListList>
                {sortedToDos.map((todo: todoprops) => todoitem(todo))}
            </ToDoListList>
            ...
        </Container>);
    }
```
Basically, you just move the functions outside the `todoitem` function but still inside the `ToDoList component` so that the functions inside the `.map()` are not repeatedly created.

Reply:
completed.